File: src/main/java/com/cout970/computer2/items/ItemBase.java
package com.cout970.computer2.items;

import com.cout970.computer2.ComputerMod;
import net.darkaqua.blacksmith.api.common.item.defaults.DefaultItemDefinition;
import net.darkaqua.blacksmith.api.client.render.item.IItemModelProvider;
import net.darkaqua.blacksmith.api.client.render.item.defaults.PlaneItemModelProvider;
import net.darkaqua.blacksmith.api.common.util.ResourceReference;

/**
 * Created by cout970 on 08/02/2016.
 */
public class ItemBase extends DefaultItemDefinition {

    public ItemBase(String name) {
        super(name);
    }

    public IItemModelProvider getModelProvider() {
        return new PlaneItemModelProvider(ComputerMod.MOD_IDENTIFIER, new ResourceReference(ComputerMod.MOD_ID, "items/" + name.toLowerCase()));
    }
}

File: src/main/kotlin/com/cout970/computer/item/ItemBase.kt
package com.cout970.computer.item

import com.cout970.computer.misc.CreativeTab
import com.cout970.computer.misc.MOD_ID
import com.cout970.computer.util.resource
import net.minecraft.item.Item

/**
 * Created by cout970 on 19/05/2016.
 */
abstract class ItemBase(registryName: String) : Item() {

    init {
        this.registryName = resource(registryName)
        this.unlocalizedName = "$MOD_ID.$registryName"
        creativeTab = CreativeTab
    }
}